src/PointwiseFunctions/GeneralRelativity/Psi4.cpp

```diff
@@ -0,0 +1,144 @@
+                       ::Tags::TempII<0, 3, Frame>>>
+      temp_buffer{get<0>(inertial_coords).size()};
+  auto& magnitude_cartesian = get<::Tags::TempScalar<0>>(temp_buffer);
+  magnitude(make_not_null(&magnitude_cartesian), inertial_coords,
+            spatial_metric);
+  auto& r_hat = get<::Tags::TempI<0, 3, Frame>>(temp_buffer);
+  for (size_t j = 0; j < 3; j++) {
+    for (size_t i = 0; i < get(magnitude_cartesian).size(); i++) {
+      if (magnitude_cartesian.get()[i] != 0.0) {
+        r_hat.get(j)[i] =
+            inertial_coords.get(j)[i] / magnitude_cartesian.get()[i];
+      } else {
+        r_hat.get(j)[i] = 0.0;
+      }
+    }
+  }
+  auto& lower_r_hat = get<::Tags::Tempi<0, 3, Frame>>(temp_buffer);
+  tenex::evaluate<ti::i>(make_not_null(&lower_r_hat),
+                         r_hat(ti::J) * spatial_metric(ti::i, ti::j));
+  auto& projection_tensor = get<::Tags::Tempii<0, 3, Frame>>(temp_buffer);
+  transverse_projection_operator(make_not_null(&projection_tensor),
+                                 spatial_metric, lower_r_hat);
+  auto& inverse_projection_tensor =
+      get<::Tags::TempII<0, 3, Frame>>(temp_buffer);
+  transverse_projection_operator(make_not_null(&inverse_projection_tensor),
+                                 inverse_spatial_metric, r_hat);
+  auto& projection_up_lo = get<::Tags::TempIj<0, 3, Frame>>(temp_buffer);
+  tenex::evaluate<ti::K, ti::i>(
+      make_not_null(&projection_up_lo),
+      projection_tensor(ti::i, ti::j) * inverse_spatial_metric(ti::K, ti::J));
+
+  auto& u8_plus = get<::Tags::Tempii<1, 3, Frame>>(temp_buffer);
+  gr::weyl_propagating(
+      make_not_null(&u8_plus), spatial_ricci, extrinsic_curvature,
+      inverse_spatial_metric, cov_deriv_extrinsic_curvature, r_hat,
+      inverse_projection_tensor, projection_tensor, projection_up_lo, 1.0);
+
+  auto& x_coord = get<::Tags::TempI<0, 3, Frame>>(temp_buffer);
+  x_coord.get(0) = 1.0;
+  x_coord.get(1) = x_coord.get(2) = 0.0;
+  auto& magnitude_x = get<::Tags::TempScalar<0>>(temp_buffer);
+  magnitude(make_not_null(&magnitude_x), x_coord, spatial_metric);
+  auto& x_hat = get<::Tags::TempI<0, 3, Frame>>(temp_buffer);
+  tenex::evaluate<ti::I>(make_not_null(&x_hat), x_coord(ti::I) / magnitude_x());
+
+  Variables<tmpl::list<::Tags::TempI<0, 3, Frame, ComplexDataVector>,
+                       ::Tags::TempI<1, 3, Frame, ComplexDataVector>>>
+      y_hat_buffer{get<0>(inertial_coords).size()};
+  auto& y_coord = get<::Tags::TempI<1, 3, Frame>>(temp_buffer);
+  y_coord.get(1) = 1.0;
+  y_coord.get(0) = y_coord.get(2) = 0.0;
+  auto& magnitude_y = get<::Tags::TempScalar<0>>(temp_buffer);
+  magnitude(make_not_null(&magnitude_y), y_coord, spatial_metric);
+  const std::complex<double> imag = std::complex<double>(0.0, 1.0);
+  auto& y_hat =
+      get<::Tags::TempI<0, 3, Frame, ComplexDataVector>>(y_hat_buffer);
+  // std::complex<double> + DataVector -> ComplexDataVector
```

Comment:
Sorry, what is this note for? The next line doesn't do any addition. It might just be left over from before and can be deleted, or maybe it needs to be changed to `*`?

Reply:
Oh whoops, yeah that should be a * this was when Alex Macedo and I were trying to figure out if tenex should handle a complex * real = complex. It's probably not needed anymore